[MESSENGER] Async emails

Guikingone opened the pull request:
Small addition in order to explain how the messenger can be used in order to send emails asynchronously using a message broker.

Need to be reviewed for syntax and sentences.

Thanks to @sroze for the help 😄

silverbackdan committed:
! prefix referencing templates

xabbuh committed:
some tweaks for the Messenger component docs

javiereguiluz committed:
Updated the embed controllers article

javiereguiluz committed:
Removed the full config dumps from config reference articles

javiereguiluz committed:
More fixes about Twig classes namespaces

javiereguiluz committed:
Removed more PHP template examples

javiereguiluz committed:
Added the versionadded directive

javiereguiluz committed:
Moved PHP-related code from Forms to the PHP templating page

javiereguiluz committed:
Replace entry/section by "article"

weaverryan committed:
[WCM] Updating the Encore documentation + New recipe

javiereguiluz committed:
Reworded the explanation

javiereguiluz committed:
Fixed Encore dir paths

javiereguiluz committed:
Mentioned how to redirect and maintain the query string

royklutman committed:
Remove unnecessary semicolons

Guikingone committed:
feat(emails): async emails

Small addition in order to explain how the messenger can be used in order to send emails asynchronously using a message broker.

Need to be reviewed for syntax and sentences.

Guikingone committed:
fix(syntax): fix on line

Guikingone committed:
fix(syntax): fix on indentation

OskarStark commented:
Hmm, not sure its worth to have such a big block of text, just for deactivating the spool...
It could be done using a note block

Guikingone commented:
@OskarStark Yep, I'm gonna correct this asap.

Guikingone committed:
fix(async): async emails

Fix on the bloc name, moved to a tip.

OskarStark commented on messenger.rst:
```diff
@@ -371,6 +371,45 @@ while still having them passed to their respective handler:
+    the spool as it's only called during the ``kernel.terminate`` event (which
+    is not dispatched during command call).
+
+    The first thing to change is the SwiftMailer configuration:
```
You talk about first but no second one 🤔

Guikingone committed:
feat(message): example for email message

Guikingone committed:
fix(syntax): fix on indentation

Guikingone commented:
@OskarStark Fixed with the second block about the message 😃

Guikingone committed:
fix(syntax): fix on parentheses

Small fix on the MessageBus parentheses call.

nikophil committed:
[Validator] Explained how to use array constant in annotation

pavlakis committed:
Show example on using a DateTimeImmutable object

Add an example on using DateTimeImmutable through the `input` option.

Guikingone committed:
fix(sentence): fix on first usage

Guikingone commented:
Ok, a small update has been made about the `first` usage as signaled by @OskarStark.

Lyrkan commented on messenger.rst:
```diff
@@ -371,6 +371,52 @@ while still having them passed to their respective handler:
+                    http://symfony.com/schema/dic/swiftmailer/swiftmailer-1.0.xsd">
+
+                <swiftmailer:config>
+                    <!-- Other keys expect the spool one --> 
```
```suggestion
                    <!-- Other keys except the spool one --> 
```

Lyrkan commented on messenger.rst:
```diff
@@ -371,6 +371,52 @@ while still having them passed to their respective handler:
+
+            // config/packages/messenger.php
+            $container->loadFromExtension('swiftmailer', array(
+                  // ... Other keys expect the spool one.
```
```suggestion
                  // ... Other keys except the spool one.
```

Lyrkan commented on messenger.rst:
```diff
@@ -371,6 +371,52 @@ while still having them passed to their respective handler:
 
+.. tip::
+
+    If you need to send asynchronous emails, it could be usefull to deactivate
```
```suggestion
    If you need to send asynchronous emails, it could be useful to deactivate
```

Lyrkan commented on messenger.rst:
```diff
@@ -371,6 +371,52 @@ while still having them passed to their respective handler:
+                  // ... Other keys expect the spool one.
+            ));
+            
+        Then time to send your message which contain the email content:
```
```suggestion
        Then time to send your message which contains the email content:
```

Lyrkan committed:
fix(wording): fix on spool exception

Co-Authored-By: Guikingone <guillaume.loulier@hotmail.fr>

Lyrkan committed:
fix(wording): fix on spool exception (php version)

Co-Authored-By: Guikingone <guillaume.loulier@hotmail.fr>

Lyrkan committed:
fix(wording): fix on spool usage

Co-Authored-By: Guikingone <guillaume.loulier@hotmail.fr>

Lyrkan committed:
fix(wording): fix on message content

Co-Authored-By: Guikingone <guillaume.loulier@hotmail.fr>

Guikingone commented:
@javiereguiluz It seems that SfCloud fails to deploy the actual revision, normal situation?

javiereguiluz commented:
@Guikingone it's not normal ... so let's ping @tucksaun to see if he can take a look. Thanks!

tucksaun commented:
@Guikingone this PR is quite old so this branch does not contain the required files for SymfonyCloud. Try to rebase onto the latest 4.1 commit and it should work.

dunglas committed:
Real-time capabilities: add docs for the Mercure integration

javiereguiluz committed:
Added the docs for the \"groups\" constraint option

javiereguiluz committed:
minor #11091 Remove `allowNull` option (julienfalque)

This PR was merged into the 4.2 branch.

Discussion
----------

Remove `allowNull` option

The option `allowNull` for constraint `NotBlank` is introduced in Symfony 4.3 but appears in the documentation of version 4.2, which can be confusing.

Commits
-------

c688d2633 Remove `allowNull` option

javiereguiluz committed:
Remove Symfony 4.3 features from 4.2 docs

javiereguiluz committed:
minor #11093 Remove Symfony 4.3 features from 4.2 docs (javiereguiluz)

This PR was merged into the 4.2 branch.

Discussion
----------

Remove Symfony 4.3 features from 4.2 docs

This continues #11091.

Commits
-------

f59dcc67c Remove Symfony 4.3 features from 4.2 docs

OskarStark committed:
wording fixes

javiereguiluz committed:
Improved the explanation about the forward method

javiereguiluz committed:
minor #11054 Added caution-box about special characters (ThomasLandauer)

This PR was merged into the 4.2 branch.

Discussion
----------

Added caution-box about special characters

Copied caution-box about special characters from https://symfony.com/doc/current/doctrine.html#configuring-the-database

Commits
-------

52bc2b965 Update email.rst

javiereguiluz committed:
Minor tweaks

javiereguiluz committed:
minor #11096 Improved the explanation about the forward method (javiereguiluz)

This PR was merged into the 4.2 branch.

Discussion
----------

Improved the explanation about the forward method

This fixes #11095 differently.

Commits
-------

ecaa7da17 Improved the explanation about the forward method

javiereguiluz committed:
minor #11084 Real-time capabilities: add docs for the Mercure Component and Bundle (dunglas)

This PR was merged into the 4.2 branch.

Discussion
----------

Real-time capabilities: add docs for the Mercure Component and Bundle

Docs for https://github.com/symfony/mercure-bundle and https://github.com/symfony/mercure.

Commits
-------

9c3107303 Real-time capabilities: add docs for the Mercure integration

Commit:
Some cache files might be written in runtime

javiereguiluz committed:
Improved the highlighting of JS examples in Mercure docs

dunglas committed:
[Mercure] Fix list rendering

javiereguiluz committed:
minor #11099 Improved the highlighting of JS examples in Mercure docs (javiereguiluz)

This PR was merged into the 4.2 branch.

Discussion
----------

Improved the highlighting of JS examples in Mercure docs

I was wrong and I told Kévin to include `<script>` in the JS code samples ... this prevents highlighting from working. See https://symfony.com/doc/current/mercure.html#subscribing

Commits
-------

95ac9549b Improved the highlighting of JS examples in Mercure docs

javiereguiluz committed:
minor #11100 [Mercure] Fix list rendering (dunglas)

This PR was merged into the 4.2 branch.

Discussion
----------

[Mercure] Fix list rendering

Fixes a tiny rendering flitch.

Commits
-------

ce1444466 [Mercure] Fix list rendering

javiereguiluz committed:
minor #11094 small wording fixes (OskarStark)

This PR was merged into the 3.4 branch.

Discussion
----------

small wording fixes

Thanks @HeahDude I applied your comments from my other PR.

Commits
-------

f0a792648 wording fixes

javiereguiluz committed:
Minor tweak

javiereguiluz committed:
minor #11092 Added the docs for the "groups" constraint option (javiereguiluz)

This PR was squashed before being merged into the 3.4 branch (closes #11092).

Discussion
----------

Added the docs for the "groups" constraint option

This continues #9557.

Question to reviewers: does this option apply to Valid and All constraints too?

Commits
-------

da9aa0e8e Added the docs for the \"groups\" constraint option

javiereguiluz committed:
Merge branch '3.4' into 4.2

* 3.4:
  Added the docs for the \"groups\" constraint option
  fixed paragraph about which versionadded directives will be removed

javiereguiluz committed:
Sort all validator constraints options

javiereguiluz committed:
minor #11101 Sort all validator constraints options (javiereguiluz)

This PR was squashed before being merged into the 4.2 branch (closes #11101).

Discussion
----------

Sort all validator constraints options

This continues #11092 and: simplifies RST tables, sorts option names and option descriptions (and fix some minor issues like missing option names in the tables).

It's impossible to review (because git doesn't detect correctly all the moved contents), so I'm going to merge it.

Commits
-------

34d8337f4 Sort all validator constraints options

javiereguiluz committed:
minor #10962 Show example on using a DateTimeImmutable object (pavlakis)

This PR was submitted for the 4.2 branch but it was merged into the 3.4 branch instead (closes #10962).

Discussion
----------

Show example on using a DateTimeImmutable object

Add an example on using DateTimeImmutable through the `input` option.

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/roadmap for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `master` for features of unreleased versions).

-->

Commits
-------

3bccc627e Show example on using a DateTimeImmutable object

javiereguiluz committed:
Minor tweaks

javiereguiluz committed:
minor #10868 [Validator] Fix `@Valid` regarding property `traverse`  (corphi)

This PR was merged into the 3.4 branch.

Discussion
----------

[Validator] Fix `@Valid` regarding property `traverse`

While researching for symfony/symfony#27090, the constraint reference turned out to be incorrect.

Commits
-------

6343b24a6 Describe actual traversal behaviour

javiereguiluz committed:
Merge branch '3.4' into 4.2

* 3.4:
  Show example on using a DateTimeImmutable object
  Describe actual traversal behaviour

javiereguiluz committed:
minor #10810 [Validator] Explained how to use array constant in annotation (nikophil)

This PR was squashed before being merged into the 3.4 branch (closes #10810).

Discussion
----------

[Validator] Explained how to use array constant in annotation

Hi,

few days ago i tried to add a new feature to the `choice` validator constraint, by adding a new `choicesFromConstant` option.
https://github.com/symfony/symfony/pull/29658
But then, i realized this was already supported.

I think this should be documented, as i think a lot of people don't know this feature.

thanks.

Commits
-------

7774aab5b [Validator] Explained how to use array constant in annotation

javiereguiluz committed:
Merge branch '3.4' into 4.2

* 3.4:
  [Validator] Explained how to use array constant in annotation

dunglas committed:
[Mercure] Link to the English version of MDN

tdutrion committed:
Fix wrong parentheses match in documentation

javiereguiluz committed:
minor #11105 Fix wrong parentheses match in documentation (tdutrion)

This PR was merged into the 4.2 branch.

Discussion
----------

Fix wrong parentheses match in documentation

There is a typo in the documentation, a squared bracket is opened, and then a bracket was closed.

Commits
-------

67eb2564d Fix wrong parentheses match in documentation

javiereguiluz committed:
minor #11104 [Mercure] Link to the English version of MDN (dunglas)

This PR was merged into the 4.2 branch.

Discussion
----------

[Mercure] Link to the English version of MDN

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/roadmap for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `master` for features of unreleased versions).

-->

Commits
-------

53f9fbe5f [Mercure] Link to the English version of MDN

xabbuh committed:
do not mention the internal ControllerTrait

xabbuh committed:
remove documentation for features introduced in 4.3

javiereguiluz committed:
Fixed some misc issues

xabbuh committed:
[#10258] fix a minor typo

javiereguiluz committed:
minor #11109 Fixed some misc issues (javiereguiluz)

This PR was merged into the 3.4 branch.

Discussion
----------

Fixed some misc issues

This replaces #10711 making only the changes that are clearly needed.

Commits
-------

1b7318cfb Fixed some misc issues

javiereguiluz committed:
Merge branch '3.4' into 4.2

* 3.4:
  Fixed some misc issues

javiereguiluz committed:
Fixed the "applies to" of the Callback constraint

javiereguiluz committed:
minor #11106 do not mention the internal ControllerTrait (xabbuh)

This PR was merged into the 4.2 branch.

Discussion
----------

do not mention the internal ControllerTrait

see https://github.com/symfony/symfony-docs/pull/11096#discussion_r263061918

Commits
-------

9d29c3939 do not mention the internal ControllerTrait

javiereguiluz committed:
minor #11107 remove documentation for features introduced in 4.3 (xabbuh)

This PR was merged into the 4.2 branch.

Discussion
----------

remove documentation for features introduced in 4.3

continuation of #11093

Commits
-------

37ba713e4 remove documentation for features introduced in 4.3

javiereguiluz committed:
Minor fix

javiereguiluz committed:
minor #11110 Fixed the "applies to" of the Callback constraint (javiereguiluz)

This PR was merged into the 3.4 branch.

Discussion
----------

Fixed the "applies to" of the Callback constraint

I've checked the source code of all constraint validators to see if the "applies to" was correct ... there was only one issue.

Commits
-------

8bba58a33 Fixed the "applies to" of the Callback constraint

javiereguiluz committed:
Merge branch '3.4' into 4.2

* 3.4:
  Fixed the "applies to" of the Callback constraint

xabbuh committed:
[#10554] fix a minor typo

xabbuh committed:
fix removing Dotenv from require-dev

javiereguiluz committed:
minor #11112 fix removing Dotenv from require-dev (xabbuh)

This PR was merged into the 4.2 branch.

Discussion
----------

fix removing Dotenv from require-dev

fixes https://github.com/symfony/symfony-docs/pull/10548#discussion_r228190470

Commits
-------

b33d8c401 fix removing Dotenv from require-dev

weaverryan committed:
adding docs about configurable flex paths

Commit:
Added a note about virtual machine for local build

althaus committed:
Add CORS_ALLOWED_ORIGINS to running mercure

I had some trouble to get the example running due to the simple fact that running mercure without a proper `CORS_ALLOWED_ORIGINS` value was preventing Firefox from connecting to the EventStream (instant close). Using chrome which keeps the connection but complains than with a proper error message hinting the to CORS_ALLOWED_ORIGIN finally helped me to figure out the problem.

So just adding the value to the example could reduce some headache.

javiereguiluz committed:
Revamped the main security reference article

xabbuh committed:
[#10689] some minor tweaks

GregOriol committed:
Add missing bracket on a php code example

OskarStark committed:
moved XSD to HTTPS

OskarStark committed:
use .. instead of .. code-block: php

javiereguiluz committed:
minor #11122 moved XSD to HTTPS (OskarStark)

This PR was merged into the 3.4 branch.

Discussion
----------

moved XSD to HTTPS

as @fabpot did this in the code base, I think we should reflect the changes in the docs, too.

Refs https://github.com/symfony/symfony/pull/30434

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/roadmap for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `master` for features of unreleased versions).

-->

Commits
-------

f57eabeef moved XSD to HTTPS

javiereguiluz committed:
Merge branch '3.4' into 4.2

* 3.4:
  moved XSD to HTTPS

javiereguiluz committed:
minor #11125 use .. instead of .. code-block: php (OskarStark)

This PR was merged into the 3.4 branch.

Discussion
----------

use .. instead of .. code-block: php

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/roadmap for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `master` for features of unreleased versions).

-->
cc @javiereguiluz

Commits
-------

ab4f29b30 use .. instead of .. code-block: php

javiereguiluz committed:
Merge branch '3.4' into 4.2

* 3.4:
  use .. instead of .. code-block: php

OskarStark committed:
use :: instead of .. code-block:: php

javiereguiluz committed:
minor #11119 Add missing bracket on a php code example (GregOriol)

This PR was merged into the 4.2 branch.

Discussion
----------

Add missing bracket on a php code example

This fixes a typo that seems to have appeared in the docs v4.2.

Commits
-------

adf8111c6 Add missing bracket on a php code example

shreypuranik committed:
Update security_checker.rst

shreypuranik committed:
Update security_checker.rst

OskarStark committed:
use terminal instead of bash

the most code blocks use terminal

javiereguiluz committed:
Minor reword

javiereguiluz committed:
minor #11131 use terminal instead of bash (OskarStark)

This PR was merged into the 3.4 branch.

Discussion
----------

use terminal instead of bash

the most code blocks use terminal

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/roadmap for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `master` for features of unreleased versions).

-->

Commits
-------

1528faca2 use terminal instead of bash

javiereguiluz committed:
Merge branch '3.4' into 4.2

* 3.4:
  use terminal instead of bash

shreypuranik committed:
Update security_checker.rst

javiereguiluz committed:
minor #11116 Add CORS_ALLOWED_ORIGINS to running mercure (althaus)

This PR was merged into the 4.2 branch.

Discussion
----------

Add CORS_ALLOWED_ORIGINS to running mercure

I had some trouble to get the example running due to the simple fact that running mercure without a proper `CORS_ALLOWED_ORIGINS` value was preventing Firefox from connecting to the EventStream (instant close). Using chrome which keeps the connection but complains than with a proper error message hinting the to CORS_ALLOWED_ORIGIN finally helped me to figure out the problem.

So just adding the value to the example could reduce some headache.

Cheers
Matthias

Commits
-------

365a6693a Add CORS_ALLOWED_ORIGINS to running mercure

javiereguiluz committed:
minor #11129 use :: instead of .. code-block:: php (OskarStark)

This PR was merged into the 3.4 branch.

Discussion
----------

use :: instead of .. code-block:: php

Follow up of https://github.com/symfony/symfony-docs/pull/11125

cc @javiereguiluz

Commits
-------

d6cbee9dd use :: instead of .. code-block:: php

javiereguiluz committed:
Merge branch '3.4' into 4.2

* 3.4:
  use :: instead of .. code-block:: php

shreypuranik committed:
Update security_checker.rst

weaverryan committed:
Making sentence not show up in code block

javiereguiluz committed:
minor #11133 Making sentence not show up in code block (weaverryan)

This PR was merged into the 4.2 branch.

Discussion
----------

Making sentence not show up in code block

See the last code block here: https://symfony.com/doc/current/configuration/dot-env-changes.html

The last sentence is being included in the code block. I'm not sure if this will fix it, but I believe it will - it changes the indentation from 4 spaces (which apparently was enough to be included in the code block) to 3, which lines it up with how we usually "continue bullet points".

Commits
-------

99dfc304b Making sentence not show up in code block

javiereguiluz committed:
minor #11115 Added a note about virtual machine for local build (Jules Pietri)

This PR was merged into the 3.4 branch.

Discussion
----------

Added a note about virtual machine for local build

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/roadmap for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `master` for features of unreleased versions).

-->

Commits
-------

091bb2205 Added a note about virtual machine for local build

javiereguiluz committed:
Minor tweak

javiereguiluz committed:
Merge branch '3.4' into 4.2

* 3.4:
  Minor tweak
  Added a note about virtual machine for local build

javiereguiluz committed:
minor #11114 adding docs about configurable flex paths (weaverryan)

This PR was submitted for the 4.2 branch but it was merged into the 3.4 branch instead (closes #11114).

Discussion
----------

adding docs about configurable flex paths

Hi friends!

Flex is a bit configurable, but it wasn't documented anywhere.

See https://twitter.com/beausimensen/status/1101863042797789192

Commits
-------

d307157bb adding docs about configurable flex paths

javiereguiluz committed:
Merge branch '3.4' into 4.2

* 3.4:
  adding docs about configurable flex paths

Nyholm committed:
Use nyholm/psr17 instead of deprecated Zend Diactoros

jt2k committed:
Fix typo

javiereguiluz committed:
minor #11136 Minor fix to OPcache regeneration instructions (jt2k)

This PR was merged into the 3.4 branch.

Discussion
----------

Minor fix to OPcache regeneration instructions

Commits
-------

8cf7481c7 Fix typo

javiereguiluz committed:
minor #11135 Use nyholm/psr17 instead of deprecated Zend Diactoros (Nyholm)

This PR was submitted for the master branch but it was merged into the 3.4 branch instead (closes #11135).

Discussion
----------

Use nyholm/psr17 instead of deprecated Zend Diactoros

The [DiactorosFactory](https://github.com/symfony/psr-http-message-bridge/blob/master/Factory/DiactorosFactory.php) is deprecated. We should update our docs to use a "pure" PSR-17 factory.

Related to: https://github.com/sensiolabs/SensioFrameworkExtraBundle/pull/608

Commits
-------

dc361b9e8 Use nyholm/psr17 instead of deprecated Zend Diactoros

javiereguiluz committed:
Minor tweaks

javiereguiluz committed:
minor #11130 Update security_checker.rst (shreypuranik, javiereguiluz)

This PR was submitted for the 4.2 branch but it was merged into the 3.4 branch instead (closes #11130).

Discussion
----------

Update security_checker.rst

Adding a minor addition to the Tip note about integrating security checker with continuous integration workflow.

Commits
-------

a86d9078a Update security_checker.rst
3cff312bd Update security_checker.rst
588aca473 Minor reword
fc77ac48d Update security_checker.rst
0f61cfcf1 Update security_checker.rst

javiereguiluz committed:
Merge branch '3.4' into 4.2

* 3.4:
  Update security_checker.rst
  Use nyholm/psr17 instead of deprecated Zend Diactoros

OskarStark committed:
use :: instead of .. code-block:: php

and removed php open tag

OskarStark committed:
removed blank lines after filepath

in xml code blocks to be consistent throughout the docs

shreypuranik committed:
Update security_checker.rst

javiereguiluz committed:
minor #11137 use :: instead of .. code-block:: php (OskarStark)

This PR was merged into the 3.4 branch.

Discussion
----------

use :: instead of .. code-block:: php

and removed php open tag

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/roadmap for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `master` for features of unreleased versions).

-->

Commits
-------

ffabe9027 use :: instead of .. code-block:: php

javiereguiluz committed:
minor #11138 removed blank lines after filepath (OskarStark)

This PR was merged into the 3.4 branch.

Discussion
----------

removed blank lines after filepath

in xml code blocks to be consistent throughout the docs

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/roadmap for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `master` for features of unreleased versions).

-->

Commits
-------

bb7914d55 removed blank lines after filepath

javiereguiluz committed:
minor #11139 Update security_checker.rst (shreypuranik)

This PR was squashed before being merged into the 3.4 branch (closes #11139).

Discussion
----------

Update security_checker.rst

Adding a small note advising users to always use the latest version of the command, as old versions will now break existing build plans because change of composer.lock checking tool.

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/roadmap for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `master` for features of unreleased versions).

-->

Commits
-------

668b4f8f0 Update security_checker.rst

javiereguiluz committed:
Reword

javiereguiluz committed:
Merge branch '3.4' into 4.2

* 3.4:
  Update security_checker.rst
  removed blank lines after filepath
  use :: instead of .. code-block:: php

OskarStark committed:
remove blank lines after file path

to be consistent throughout the docs

OskarStark committed:
added missing php before bin/console

to be consistent throughout the docs

OskarStark committed:
remove space before self closing xml tag

to be consistent throughout the docs

juuuuuu committed:
Update access_control.rst role

I think that there is an error about the second rule's `role` value. It should be `ROLE_USER_PORT` instead of `ROLE_USER_IP`.
Correct me if I'm wrong!

javiereguiluz committed:
minor #11141 added missing php before bin/console (OskarStark)

This PR was merged into the 3.4 branch.

Discussion
----------

added missing php before bin/console

to be consistent throughout the docs

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/roadmap for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `master` for features of unreleased versions).

-->

Commits
-------

a8afae4d4 added missing php before bin/console

javiereguiluz committed:
minor #11140 remove blank lines after file path (OskarStark)

This PR was merged into the 3.4 branch.

Discussion
----------

remove blank lines after file path

to be consistent throughout the docs

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/roadmap for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `master` for features of unreleased versions).

-->

Commits
-------

bed07543e remove blank lines after file path

javiereguiluz committed:
Merge branch '3.4' into 4.2

* 3.4:
  added missing php before bin/console
  remove blank lines after file path

javiereguiluz committed:
minor #11143 Update access_control.rst role (juuuuuu)

This PR was merged into the 4.2 branch.

Discussion
----------

Update access_control.rst role

I think that there is an error about the second rule's `role` value. It should be `ROLE_USER_PORT` instead of `ROLE_USER_IP`.
Correct me if I'm wrong!

These changes don't need to be done for others versions.

Commits
-------

b4f4ceb8b Update access_control.rst role

OskarStark committed:
remove .. code-block:: php

and replace it by a text

javiereguiluz committed:
minor #11146 remove .. code-block:: php (OskarStark)

This PR was submitted for the master branch but it was merged into the 3.4 branch instead (closes #11146).

Discussion
----------

remove .. code-block:: php

and replace it by a text

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/roadmap for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `master` for features of unreleased versions).

-->

Commits
-------

b9edc206e remove .. code-block:: php

javiereguiluz committed:
Minor tweak

javiereguiluz committed:
Merge branch '3.4' into 4.2

* 3.4:
  Minor tweak
  remove .. code-block:: php

xabbuh committed:
do not mock the validator in form type tests

ThePeterMick committed:
updating spacing

Krap committed:
Added php opening tag to sample controller

It took me a while to figure out why the sample didn't seem to work, despite being experimented in php ; i checked many times namespace, uses and class declaration before finding that i just missed the opening tag. Think it would help others (even more if unexperimented) discovering symfony by copy/pasting sample code from the doc !

OskarStark committed:
use :: instead of .. code-block:: php

javiereguiluz committed:
minor #11150 updating spacing (ptrm04)

This PR was merged into the 3.4 branch.

Discussion
----------

updating spacing

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/roadmap for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `master` for features of unreleased versions).

-->

Commits
-------

33e7a063d updating spacing

javiereguiluz committed:
minor #11153 use :: instead of .. code-block:: php (OskarStark)

This PR was merged into the 3.4 branch.

Discussion
----------

use :: instead of .. code-block:: php

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/roadmap for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `master` for features of unreleased versions).

-->

Commits
-------

60eeca0b2 use :: instead of .. code-block:: php

javiereguiluz committed:
Minor tweaks

javiereguiluz committed:
minor #11149 do not mock the validator in form type tests (xabbuh)

This PR was merged into the 3.4 branch.

Discussion
----------

do not mock the validator in form type tests

Commits
-------

272e47970 do not mock the validator in form type tests

javiereguiluz committed:
Merge branch '3.4' into 4.2

* 3.4:
  Minor tweaks
  use :: instead of .. code-block:: php
  do not mock the validator in form type tests
  updating spacing

javiereguiluz committed:
minor #11151 Added php opening tag to sample controller (Krap)

This PR was submitted for the 4.2 branch but it was merged into the 3.4 branch instead (closes #11151).

Discussion
----------

Added php opening tag to sample controller

It took me a while to figure out why the sample didn't seem to work, despite being experimented in php ; i checked many times namespace, uses and class declaration before finding that i just missed the opening tag. Think it would help others (even more if unexperimented) discovering symfony by copy/pasting sample code from the doc !

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/roadmap for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `master` for features of unreleased versions).

-->

Commits
-------

4c3a0d040 Added php opening tag to sample controller

javiereguiluz committed:
Merge branch '3.4' into 4.2

* 3.4:
  Added php opening tag to sample controller

javiereguiluz committed:
Added the missing annotations to a routing example

javiereguiluz committed:
Added Oskar to the Docs Team

samnela committed:
[Mercure] Fix link to high-level implementations

javiereguiluz committed:
minor #11162 Added Oskar to the Docs Team (javiereguiluz)

This PR was merged into the 3.4 branch.

Discussion
----------

Added Oskar to the Docs Team

🎉

Commits
-------

b682cc471 Added Oskar to the Docs Team

javiereguiluz committed:
Don't use AbstractController in 3.4 branch

tyx committed:
Fix stateless mention

According to https://github.com/symfony/symfony/blob/5aa0967f9f0ab803ededefb040d48a0ebc7a27a6/src/Symfony/Component/Security/Guard/GuardAuthenticatorHandler.php#L43 Firewall of pure API should have `stateless` to `true`  in fact

javiereguiluz committed:
minor #11165 Don't use AbstractController in 3.4 branch (javiereguiluz)

This PR was merged into the 3.4 branch.

Discussion
----------

Don't use AbstractController in 3.4 branch

In 3.4 we always extend from Controller instead of AbstractController, so this PR fixes the wrong occurrences. The only remaining occurrence is one which explicitly explains something about AbstractController.

Commits
-------

a464d7e0e Don't use AbstractController in 3.4 branch

javiereguiluz committed:
Merge branch '3.4' into 4.2

* 3.4:
  Don't use AbstractController in 3.4 branch
  Added Oskar to the Docs Team

javiereguiluz committed:
minor #11159 Added the missing annotations to a routing example (javiereguiluz)

This PR was merged into the 3.4 branch.

Discussion
----------

Added the missing annotations to a routing example

Commits
-------

e18236c84 Added the missing annotations to a routing example

javiereguiluz committed:
Merge branch '3.4' into 4.2

* 3.4:
  Added the missing annotations to a routing example

javiereguiluz committed:
Updated code for Symfony 4.x

OskarStark committed:
promote AbstractController in 4.2 branch

OskarStark committed:
typo in comment (php example)

xabbuh committed:
regroup some constraints

OskarStark committed:
missing brackets

wouterj committed:
minor #11173 regroup some constraints (xabbuh)

This PR was merged into the 3.4 branch.

Discussion
----------

regroup some constraints

fixes #11167

Commits
-------

5e4794881 regroup some constraints

wouterj committed:
minor #11172 typo in comment (php example) (OskarStark)

This PR was merged into the 3.4 branch.

Discussion
----------

typo in comment (php example)

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/roadmap for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `master` for features of unreleased versions).

-->

Commits
-------

dfd369922 typo in comment (php example)

wouterj committed:
minor #11171 promote AbstractController in 4.2 branch (OskarStark)

This PR was merged into the 4.2 branch.

Discussion
----------

promote AbstractController in 4.2 branch

Follow up of @javiereguiluz's https://github.com/symfony/symfony-docs/pull/11165

Commits
-------

b6ab4fae3 promote AbstractController in 4.2 branch

wouterj committed:
bug #11164 [Mercure] Fix link to high-level implementations (samnela)

This PR was merged into the 4.2 branch.

Discussion
----------

[Mercure] Fix link to high-level implementations

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/roadmap for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `master` for features of unreleased versions).

-->

Commits
-------

8bc40ea08 [Mercure] Fix link to high-level implementations

wouterj committed:
bug #11166 [Security] Fix stateless mention (tyx)

This PR was merged into the 4.2 branch.

Discussion
----------

[Security] Fix stateless mention

According to https://github.com/symfony/symfony/blob/5aa0967f9f0ab803ededefb040d48a0ebc7a27a6/src/Symfony/Component/Security/Guard/GuardAuthenticatorHandler.php#L43 Firewall of pure API should have `stateless` to `true`  in fact

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/roadmap for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `master` for features of unreleased versions).

-->

Commits
-------

2283cd214 Fix stateless mention

wouterj committed:
Merge branch '3.4' into 4.2

* 3.4:
  regroup some constraints
  typo in comment (php example)

wouterj committed:
minor #11142 remove space before self closing xml tag (OskarStark)

This PR was merged into the 3.4 branch.

Discussion
----------

remove space before self closing xml tag

to be consistent throughout the docs

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/roadmap for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `master` for features of unreleased versions).

-->

Commits
-------

d14992116 remove space before self closing xml tag

wouterj committed:
Merge branch '3.4' into 4.2

* 3.4:
  remove space before self closing xml tag

OskarStark committed:
remove space before self closing xml tag

wouterj committed:
minor #11175 remove space before self closing xml tag (OskarStark)

This PR was merged into the 4.2 branch.

Discussion
----------

remove space before self closing xml tag

@wouterj this one is following your [comment](https://github.com/symfony/symfony-docs/pull/11142#issuecomment-473575680)

Commits
-------

db87ab539 remove space before self closing xml tag

wouterj committed:
minor #11176 use correct quote (OskarStark)

This PR was merged into the 4.2 branch.

Discussion
----------

use correct quote

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/roadmap for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `master` for features of unreleased versions).

-->

Commits
-------

a5f792924 use correct quote

wouterj committed:
bug #11174 added missing brackets (OskarStark)

This PR was merged into the 3.4 branch.

Discussion
----------

added missing brackets

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/roadmap for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `master` for features of unreleased versions).

-->

Commits
-------

009931f3c missing brackets

wouterj committed:
Merge branch '3.4' into 4.2

* 3.4:
  missing brackets

zbrag committed:
Update advanced-config.rst

It took me some time to find https://github.com/symfony/webpack-encore/issues/477 , so I guess it would be better to add it to the official documentation.

OskarStark committed:
Use the \".. deprecated::\" directive for deprecations

OskarStark committed:
use same order for all configuration-blocks

ensure the following order of code-blocks in a .. configuration-block:: directive:
 - php-annotations
 - yaml
 - xml
 - php

ThomasLandauer committed:
Fixing error at Class Constraint Validator

The correct annotation is `@AcmeAssert\ProtocolClass`, not `@AcmeAssert\ProtocolClassValidator`.

The other changes are just minor rewordings.

OskarStark committed:
use :: instead of .. code-block:: php

oliverquynh committed:
Improve now what sentence

juuuuuu committed:
Update entity.rst

Highlight the `name` option and use FQCN instead of string.

javiereguiluz committed:
minor #11183 Update entity.rst (juuuuuu)

This PR was submitted for the 4.2 branch but it was merged into the 3.4 branch instead (closes #11183).

Discussion
----------

Update entity.rst

Highlight the `name` property and use FQCN instead of string.

Commits
-------

1995aa118 Update entity.rst

javiereguiluz committed:
Merge branch '3.4' into 4.2

* 3.4:
  Update entity.rst

javiereguiluz committed:
minor #11177 Update advanced-config.rst to contain more information about multi encore-builds (zbrag)

This PR was submitted for the 4.2 branch but it was merged into the 3.4 branch instead (closes #11177).

Discussion
----------

Update advanced-config.rst to contain more information about multi encore-builds

It took me some time to find https://github.com/symfony/webpack-encore/issues/477 , so I guess it would be better to add it to the official documentation.

Commits
-------

954b3985c Update advanced-config.rst

javiereguiluz committed:
Tweaks

javiereguiluz committed:
Merge branch '3.4' into 4.2

* 3.4:
  Update advanced-config.rst

javiereguiluz committed:
Minor tweaks

javiereguiluz committed:
minor #11180 Fixing error at Class Constraint Validator (ThomasLandauer)

This PR was merged into the 3.4 branch.

Discussion
----------

Fixing error at Class Constraint Validator

The correct annotation is `@AcmeAssert\ProtocolClass`, not `@AcmeAssert\ProtocolClassValidator`.

The other changes are just minor rewordings.

Commits
-------

0b6c6f93c Fixing error at Class Constraint Validator

javiereguiluz committed:
Minor tweak

javiereguiluz committed:
Merge branch '3.4' into 4.2

* 3.4:
  Fixing error at Class Constraint Validator

favianioel committed:
Fix typo in http_kernel_controller_resolver.rst

oliverquynh committed:
Correct a sentence meaning

Commit:
Add precision on the concerned fields

jmsche committed:
[Encore] Fixed typo

OskarStark committed:
fix indention

javiereguiluz committed:
minor #11191 fix indention (OskarStark)

This PR was merged into the 3.4 branch.

Discussion
----------

fix indention

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/roadmap for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `master` for features of unreleased versions).

-->

Commits
-------

6f18433d6 fix indention

javiereguiluz committed:
minor #11189 [Encore] Fixed typo (jmsche)

This PR was squashed before being merged into the 3.4 branch (closes #11189).

Discussion
----------

[Encore] Fixed typo

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/roadmap for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `master` for features of unreleased versions).

-->

Commits
-------

0973a0175 [Encore] Fixed typo

javiereguiluz committed:
minor #11184 Fix typo in http_kernel_controller_resolver.rst (favianioel)

This PR was merged into the 4.2 branch.

Discussion
----------

Fix typo in http_kernel_controller_resolver.rst

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/roadmap for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `master` for features of unreleased versions).

-->

Commits
-------

2a8b2fed6 Fix typo in http_kernel_controller_resolver.rst

javiereguiluz committed:
Merge branch '3.4' into 4.2

* 3.4:
  [Encore] Fixed typo
  fix indention

javiereguiluz committed:
minor #11186 Correct a sentence meaning (xuanquynh)

This PR was merged into the 3.4 branch.

Discussion
----------

Correct a sentence meaning

I believe that there is a mistake about passive and active sentence here. The "you are finished" clause has no meaning, doesn't it?

We can use two simple ways:
- When your work is finished,
- When you finish your work,

Commits
-------

06d399dac Correct a sentence meaning

javiereguiluz committed:
Minor tweak

javiereguiluz committed:
minor #11188  Add precision on the concerned fields in the Default validation group (florianbelhomme)

This PR was merged into the 3.4 branch.

Discussion
----------

 Add precision on the concerned fields in the Default validation group

Explicitly tells which field's constraint is in the ``Default`` validation group.

Commits
-------

dbda71008 Add precision on the concerned fields

javiereguiluz committed:
Minor tweak

javiereguiluz committed:
minor #11179 use same order for all configuration-blocks (OskarStark)

This PR was merged into the 3.4 branch.

Discussion
----------

use same order for all configuration-blocks

ensure the following order of code-blocks in a `.. configuration-block::` directive:

1. `php-annotations`
2. `yaml`
3. `xml`
4. `php`

Commits
-------

7cfa0d7e9 use same order for all configuration-blocks

javiereguiluz committed:
minor #11178 Use the ".. deprecated::" directive for deprecations (OskarStark)

This PR was squashed before being merged into the 3.4 branch (closes #11178).

Discussion
----------

Use the ".. deprecated::" directive for deprecations

Refs #11154

## Todo:

- [ ] We need no ensure the correct html markup
- [ ] @symfony/docs-team please review the new documentation about "[how/when to use this new directive](https://github.com/symfony/symfony-docs/commit/2b0504d9aeea9c523ccb5b85fb19962d25a6d198)"

Commits
-------

b48d0c77d Use the \".. deprecated::\" directive for deprecations

javiereguiluz committed:
minor #11181 use :: instead of .. code-block:: php (OskarStark)

This PR was merged into the 3.4 branch.

Discussion
----------

use :: instead of .. code-block:: php

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/roadmap for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `master` for features of unreleased versions).

-->

Commits
-------

a96cb3a85 use :: instead of .. code-block:: php

javiereguiluz committed:
Minor tweak

javiereguiluz committed:
Merge branch '3.4' into 4.2

* 3.4:
  Use the \".. deprecated::\" directive for deprecations
  use same order for all configuration-blocks
  Add precision on the concerned fields
  Correct a sentence meaning
  use :: instead of .. code-block:: php

OskarStark committed:
use correct order of code-blocks

javiereguiluz committed:
minor #11194 use correct order of code-blocks (OskarStark)

This PR was merged into the 4.2 branch.

Discussion
----------

use correct order of code-blocks

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/roadmap for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `master` for features of unreleased versions).

-->

Commits
-------

8c09d0292 use correct order of code-blocks

OskarStark committed:
Consistency changes

OskarStark committed:
typo

javiereguiluz committed:
minor #11195 Consistency changes (OskarStark)

This PR was squashed before being merged into the 4.2 branch (closes #11195).

Discussion
----------

Consistency changes

This PR brings more consistency to the 4.2 branch and should be reviewed commit by commit.

Commits
-------

167329811 Consistency changes

javiereguiluz committed:
minor #11196 use "behavior" instead of "behaviour" (OskarStark)

This PR was merged into the 4.2 branch.

Discussion
----------

use "behavior" instead of "behaviour"

We use **behavior** many more then **behaviour** throughout the documentation.

Commits
-------

27b762172 typo

javiereguiluz committed:
minor #11098 Some cache files might be written in runtime (cvdwel)

This PR was submitted for the master branch but it was merged into the 4.2 branch instead (closes #11098).

Discussion
----------

Some cache files might be written in runtime

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/roadmap for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `master` for features of unreleased versions).

-->

Commits
-------

3fcb22cab Some cache files might be written in runtime

OskarStark committed:
use consistent placeholder comments in yaml, php and xml

javiereguiluz committed:
Minor reword

javiereguiluz committed:
minor #11197 use consistent placeholder comments in yaml, php and xml (OskarStark)

This PR was merged into the 3.4 branch.

Discussion
----------

use consistent placeholder comments in yaml, php and xml

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/roadmap for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `master` for features of unreleased versions).

-->

Commits
-------

bee41916f use consistent placeholder comments in yaml, php and xml

javiereguiluz committed:
Merge branch '3.4' into 4.2

* 3.4:
  use consistent placeholder comments in yaml, php and xml

javiereguiluz committed:
minor #11117 Revamped the main security reference article (javiereguiluz)

This PR was squashed before being merged into the 3.4 branch (closes #11117).

Discussion
----------

Revamped the main security reference article

This documents the missing global security options and cleans up the existing reference. There are still a ton of missing config options ... but I'll fix that in a future PR.

Fixes #11102.

Commits
-------

f2a02af67 Revamped the main security reference article

javiereguiluz committed:
Merge branch '3.4' into 4.2

* 3.4:
  Revamped the main security reference article

favianioel committed:
Update unit_testing.rst

While following the tutorial changing the property from $controllerResolver to $resolver generated a bug because in previous example from httpkernel in the file the property was called like this 
$controller = $this->controllerResolver->getController($request);
It is not a fix but it could help the people that follow the tutorial not get stuck :+1:

geertclerx committed:
Remove unnecessary backslash escape

OskarStark committed:
add missing brackets

OskarStark committed:
use "behavior" instead of "behaviour"

javiereguiluz committed:
Fixed some references

javiereguiluz committed:
minor #11203 Fixed some references (javiereguiluz)

This PR was merged into the 4.2 branch.

Discussion
----------

Fixed some references

Commits
-------

3f646a280 Fixed some references

OskarStark committed:
removed blank line and space

xabbuh committed:
[#11139] fix a minor typo

xabbuh committed:
Merge branch '3.4' into 4.2

* 3.4:
  [#11139] fix a minor typo

xabbuh committed:
minor #11201 add missing brackets (OskarStark)

This PR was merged into the 4.2 branch.

Discussion
----------

add missing brackets

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/roadmap for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `master` for features of unreleased versions).

-->

Commits
-------

7be1bcb7d add missing brackets

xabbuh committed:
minor #11202 use "behavior" instead of "behaviour" (OskarStark)

This PR was merged into the 3.4 branch.

Discussion
----------

use "behavior" instead of "behaviour"

I provided [this PR](https://github.com/symfony/symfony-docs/pull/11196) against `4.2`,  but it should be done against `3.4`.

If there is another approach (cherry picking or sth else), feel free to close this PR 👍

Commits
-------

3b961fe97 use "behavior" instead of "behaviour"

xabbuh committed:
minor #11204 removed blank line and space (OskarStark)

This PR was merged into the 3.4 branch.

Discussion
----------

removed blank line and space

Refs #11142 #11175
Refs #11138 #11140

Commits
-------

ecceb77cd removed blank line and space

xabbuh committed:
minor #11199 Remove unnecessary backslash escape (clerx)

This PR was merged into the 3.4 branch.

Discussion
----------

Remove unnecessary backslash escape

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/roadmap for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `master` for features of unreleased versions).

-->

Commits
-------

80fec9eb0 Remove unnecessary backslash escape

xabbuh committed:
minor #11182 Improve now what sentence (xuanquynh)

This PR was merged into the 3.4 branch.

Discussion
----------

Improve now what sentence

The "Now What?" section means 2 things:
+ Tell the developers that the component is ready.
+ Tell the developers to go through a specific documentation.

Therefore, I think it should be separated into two sentences for readable reason. By the way, I just remove unnecessary word "that" and add a comma.

Commits
-------

6b6a127c6 Improve now what sentence

xabbuh committed:
minor #11198 Update unit_testing.rst (favianioel)

This PR was submitted for the 4.2 branch but it was merged into the 3.4 branch instead (closes #11198).

Discussion
----------

Update unit_testing.rst

While following the tutorial changing the property from $controllerResolver to $resolver generated a bug because in previous example from httpkernel in the file the property was called like this
$controller = $this->controllerResolver->getController($request);
It is not a fix but it could help the people that follow the tutorial not get stuck :+1:

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/roadmap for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `master` for features of unreleased versions).

-->

Commits
-------

3853c5b33 Update unit_testing.rst

xabbuh committed:
[#8416] some minor tweaks

xabbuh committed:
[#8495] some minor tweaks

jfhovinne committed:
Add missing security guard sub-component

javiereguiluz committed:
minor #11206 Add missing security guard sub-component (jfhovinne)

This PR was squashed before being merged into the 3.4 branch (closes #11206).

Discussion
----------

Add missing security guard sub-component

Introduction says "The Security component is divided into four smaller sub-components" while only three are listed.
Added security-guard.

Commits
-------

83cb4b8c3 Add missing security guard sub-component

javiereguiluz committed:
Minor tweak

javiereguiluz committed:
Merge branch '3.4' into 4.2

* 3.4:
  Add missing security guard sub-component
  [#8416] some minor tweaks
  Update unit_testing.rst
  removed blank line and space
  use "behavior" instead of "behaviour"
  Remove unnecessary backslash escape
  Improve now what sentence

pbowyer committed:
Document Transition Blockers

michaelKaefer committed:
grammar typo

ThomasLandauer committed:
Fixing minor error

It's not the *form*'s class option, but the class option of the current field. Also removed the unnecessary self-referencing link.

javiereguiluz committed:
minor #11213 grammar typo (michaelKaefer)

This PR was submitted for the 4.2 branch but it was merged into the 3.4 branch instead (closes #11213).

Discussion
----------

grammar typo

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/roadmap for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `master` for features of unreleased versions).

-->

Commits
-------

3be0563fa grammar typo

javiereguiluz committed:
minor #11216 Fixing minor error (ThomasLandauer)

This PR was merged into the 3.4 branch.

Discussion
----------

Fixing minor error

It's not the *form*'s class option, but the class option of the current field. Also removed the unnecessary self-referencing link.

Commits
-------

8c512bd30 Fixing minor error

javiereguiluz committed:
Fixed a syntax issue

javiereguiluz committed:
Merge branch '3.4' into 4.2

* 3.4:
  Fixed a syntax issue
  grammar typo
  Fixing minor error

javiereguiluz committed:
minor #11212 Document Transition Blockers (pbowyer)

This PR was submitted for the 4.1 branch but it was squashed and merged into the 4.2 branch instead (closes #11212).

Discussion
----------

Document Transition Blockers

The explanation needs expanding; the example is full-fledged.

Question: do the second and third parameters to `new TransitionBlocker()` need to be demonstrated in the example?

Commits
-------

2c640707f Document Transition Blockers

javiereguiluz committed:
Minor tweaks

javiereguiluz commented:
I'm closing this old pull request because we recently added something like this for the new mailer component. See https://symfony.com/doc/current/mailer.html#sending-messages-async  Thanks.
